### DISCUSSION TITLE
Solar API documentation

### DISCUSSION THREAD

**Discussion by SanJoseBart** [timdorr/tesla-api#581] [General]
I've found little current information on Tesla's solar APIs on the web, so I've documented the couple of endpoints I'm familiar with and submitted a pull request to add the documentation to this repository.  I hope expanding this repository to cover solar APIs is appropriate.

I've documented the endpoints `history` and `calendar_history`, for retrieving solar energy generation data from Tesla solar installations.  There's a third similar endpoint, `telemetry_history`, that I've been unable to use.  Anybody know what it does and how to use it?  I only know that the `kind` URL parameter is required, but the values used for `kind` in the other endpoints are not valid for `telemetry_history`.

Thanks!

**Comment by shred86** [comment]
@SanJoseBart Out of curiosity, what type of solar system do you have? Solar roof or solar panels? Inverter brand (e.g. Delta, Tesla, etc.)? Powerwalls (if so, what type)?

I just made a PR #606 to add some additional data for a few endpoints but I have a solar panel system with a Tesla inverter and no powerwalls. I do have a Neurio energy monitoring device.

I noticed with the `/api/1/energy_sites/{site_id}/history` endpoint using `kind='energy'` or `kind='power'`, all of my values are always 0.

Using `/api/1/energy_sites/{site_id}/calendar_history`, I am able to see actual values that match my Tesla app. However, even though I'm passing in `end_date='2022-07-28T01:00:00-07:00'`, the response shows `installation_time_zone": "UTC"` and all the timestamps are in Z.

**Comment by SanJoseBart** [comment]
Hi @shred86,

I have an older Solar City system with solar panels.  Tesla bought Solar 
City a while back.  The inverter is only labelled as Solar City, model 
PVI-3.0-outd-s-us-z-a.  There is a reference to www.abb.com on the 
label, so I suspect ABB is the original manufacturer.  No power wall.

For both the "history" and "calendar-history" endpoints, in my case all 
the returned statistics are 0 except for the generated power. 
Fortunately the generated power was all I wanted.  Of course the 
generated power is also 0 for timestamps at night.

Even when I use the "history" endpoint with no timestamp parameters, the 
returned stats have timestamps in my local timezone.  Perhaps Tesla 
obtains the timezone from the login sequence, or more likely, from my 
account details.  See if there is something in your Tesla account 
settings to specify a timezone.

Cheers,
Bart
